src/core/puppeteer.ts

```diff
@@ -206,9 +206,7 @@ async function findChrome(): Promise<string | undefined> {
       "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome",
       "/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge",
     ];
-    return programs.find((p) => {
-      safeExistsSync(p);
-    });
+    return programs.find(safeExistsSync);
```

Comment:
Why should this make a difference?

The transformation of `f` to `(parameter) => f(parameter)` makes a difference when functions take more than one parameter (so that eg `find` doesn't accidentally pass a stray parameter). But `safeExistsSync` is:

https://github.com/quarto-dev/quarto-cli/blob/a90f4b0040229318e343946323d601075edcb452/src/core/path.ts#L112-L118

That takes a single parameter. So I don't understand this.

Reply:
I don't either but for some reason the previous code does not detect/return the path when it exists and the one here does.
I am no TypeScript expert.

Reply:
The current code returns undefined.
I even try adding Brave in the array to test one more browser.
Only the proposed code/syntax returns the correct path.

Reply:
Correct me if I'm wrong but doesn't

```ts
    return programs.find((p) => {
      safeExistsSync(p);
    });
```
always fail,  because it should either have `return` or no braces? Functions return `undefined` by default. 

I.e. this is *not* the same as 

> `(parameter) => f(parameter)`

Reply:
Ah, that's _exactly_ what it is, Gordon. I always get confused by fat arrow syntax with braces.

More specifically

```js
(p) => { safeExistsSync(p); }
```

is _very different_ from

```js
(p) => safeExistsSync(p); // which _is_ equivalent to safeExistsSync
```

Ok, this all makes sense now, except that I don't really understand why TypeScript let us write the obviously-broken function before!

Reply:
I learned something. Thanks @gordonwoodhull

Reply:
Oh thanks ! I believe my mistake 😓 good catch @mcanouil ! 

I will remember the difference now... Though I get mixed up by R syntax so it could happen again 😅

Reply:
You did say you did not test macOS on the initial PR. And I believe to know this, you need to be bitten by it first^^